features/support/env.rb

```diff
@@ -1,6 +1,5 @@
 require 'simplecov'
 SimpleCov.command_name 'Cucumber'
-SimpleCov.minimum_coverage 100
```

Comment:
I think it's OK that neither suite by itself covers 100% of the code. One thing that comes to mind is that we have some code that only runs in background jobs, and it would be difficult if not impossible to write cucumber tests for that code.

Perhaps the thing do is for each suite maintain a list of classes that we expect to have 100% coverage by that suite. We would just need to make sure that all classes fall into at least one of those lists.

Reply:
That seems like (possibly) quite a long story to make that coverage work, and also a place where we could forget to add new classes in the future.

What would be great would be if we could merge the coverage reports from the two parallel runs and then check for 100% coverage.

Reply:
Sorry about that - I think I misunderstood the question. I thought the question was "do you think it's ideal that neither suite by itself covers 100% of the code?" If the question is "how can we deal with the fact that neither suite by itself covers 100% of the code?" then I agree that somehow merging the two reports would be a great way to go.

Reply:
@nickmarden , I'm also OK with full coverage only arising from a combination of RSpec and Cucumber tests. Our cucumber tests for super admin features are a good example of why we might do that. I don't trust RSpec tests to cover those, as the tests have to be far too close to the code as implemented, so I prefer cuke tests for super admin pages.

On another note, @HalahAb's plan to "split up the cucumber tests to have them run in parallel within their own job", so that the RSpec tests and the cuke tests can run in parallel should halve the run time while avoiding the hiccups resulting from out-of-order RSpec tests. That would be an excellent first step toward full parallelization.

Reply:
I think I asked my question the wrong way, so sorry about that.

I'm not suggesting that we should change are test so that both rspec and cucumber have 100% coverage independently. I was implicitly asking how we should get around that the fact that they don't - in other words, by finding some way to simulate the "combined test coverage" behavior that comes from our current approach.

Reply:
@nickmarden , re: "What would be great would be if we could merge the coverage reports from the two parallel runs and then check for 100% coverage." -- That is essentially what already happens.

The cuke tests and RSpec tests each generate a separate JSON file, which are then aggregated to get the full coverage:
```
 ./cc-test-reporter format-coverage --output coverage/codeclimate.rspec.json
...
 ./cc-test-reporter format-coverage --output coverage/codeclimate.cucumber.json
...
 ./cc-test-reporter sum-coverage coverage/codeclimate.*.json
```
See https://github.com/RapidRiverSoftware/search-gov/blob/ef9d8c3897d025d3f82e6673666cb5e60a982d96/.circleci/config.yml#L228

Reply:
...So in theory, any subset of the tests should be able to generate its own `coverage/codeclimate.whatever.json` file to contribute to the overall coverage picture.

Reply:
It would also be interesting to check out CC's native coverage checks: https://codeclimate.com/changelog/59baa212ced24c026c0017b8/

Reply:
Oh, I'm sorry. I thought this line that we're talking about was being removed because, somehow, code coverage was no longer at 100%. According to what you're saying here, that shouldn't be the case?

Reply:
That is correct - that line should stay, because we do have 100% coverage.

Reply:
Oh wait, I think I get it. It's not that we're relaxing the 100% coverage requirement, we're just not looking for it to be satisfied at the end of the feature (cucumber) run any more?

Reply:
(Sorry, I should have led off with "I agree that we definitely don't want to remove this".)

Reply:
I think we need @HalahAb to comment on the reason for the line removal. I'm wondering if she has implemented the native CC coverage, which would make this line redundant...?

Reply:
Yes, I'm getting the feeling that's what's going on :-) - we can wait for her to chime in before we proceed.

Reply:
Hi all,

Yes, the approach here is to scrap using SimpleCov to perform the coverage check, and to use Code Climate (CC) instead.

If you look at the new [Workflow for builds](https://circleci.com/workflow-run/531d69e4-14ef-4426-8885-ae34eddbf98b) you can see that the last step takes all the coverage reports from each parallel job, combines them, then submits them to CC. From there, we want CC to report to the PR that coverage is sufficient.

At the moment, CC isn't configured to report failure if coverage drops to less than 100%. @nickmarden @MothOnMars could either of you help us with that configuration, either by giving us access to CC or going in and doing it yourself?